isthmus/src/main/java/io/substrait/isthmus/expression/LiteralConstructorConverter.java

```diff
@@ -18,29 +19,53 @@ public class LiteralConstructorConverter implements CallConverter {
   static final org.slf4j.Logger logger =
       org.slf4j.LoggerFactory.getLogger(LiteralConstructorConverter.class);
 
+  private final TypeConverter typeConverter;
+
+  public LiteralConstructorConverter(TypeConverter typeConverter) {
+    this.typeConverter = typeConverter;
+  }
+
   @Override
   public Optional<Expression> convert(
       RexCall call, Function<RexNode, Expression> topLevelConverter) {
     SqlOperator operator = call.getOperator();
     if (operator instanceof SqlArrayValueConstructor) {
-      return Optional.of(
-          ExpressionCreator.list(
-              false,
-              call.operands.stream()
-                  .map(t -> ((Expression.Literal) topLevelConverter.apply(t)))
-                  .collect(java.util.stream.Collectors.toList())));
+      return call.getOperands().isEmpty()
+          ? toEmptyListLiteral(call)
+          : toNonEmptyListLiteral(call, topLevelConverter);
     } else if (operator instanceof SqlMapValueConstructor) {
-      List<Expression.Literal> literals =
-          call.operands.stream()
-              .map(t -> ((Expression.Literal) topLevelConverter.apply(t)))
-              .collect(java.util.stream.Collectors.toList());
-      Map<Expression.Literal, Expression.Literal> items = new HashMap<>();
-      assert literals.size() % 2 == 0;
-      for (int i = 0; i < literals.size(); i += 2) {
-        items.put(literals.get(i), literals.get(i + 1));
-      }
-      return Optional.of(ExpressionCreator.map(false, items));
+      return toMapLiteral(call, topLevelConverter);
     }
     return Optional.empty();
   }
+
+  private Optional<Expression> toMapLiteral(
+      RexCall call, Function<RexNode, Expression> topLevelConverter) {
+    List<Expression.Literal> literals =
+        call.operands.stream()
+            .map(t -> ((Expression.Literal) topLevelConverter.apply(t)))
+            .collect(java.util.stream.Collectors.toList());
+    Map<Expression.Literal, Expression.Literal> items = new HashMap<>();
+    assert literals.size() % 2 == 0;
+    for (int i = 0; i < literals.size(); i += 2) {
+      items.put(literals.get(i), literals.get(i + 1));
+    }
+    return Optional.of(ExpressionCreator.map(false, items));
+  }
+
+  private Optional<Expression> toNonEmptyListLiteral(
+      RexCall call, Function<RexNode, Expression> topLevelConverter) {
+    return Optional.of(
+        ExpressionCreator.list(
+            false,
+            call.operands.stream()
+                .map(t -> ((Expression.Literal) topLevelConverter.apply(t)))
+                .collect(java.util.stream.Collectors.toList())));
+  }
+
+  private Optional<Expression> toEmptyListLiteral(RexCall call) {
+    var calciteElementType = call.getType().getComponentType();
+    var substraitElementType = typeConverter.toSubstrait(calciteElementType);
+    return Optional.of(ExpressionCreator.emptyList(false, substraitElementType));
```

Comment:
Instead of setting then nullability to `false` here, should we use the nullability we get from Calcite in `call.getType.isNullable`? 

I _suspect_ it should always be `false`, but I'm not sure.